widgets/common/src/filters/FiltersDefinitionForm.tsx

```diff
@@ -1,31 +1,37 @@
 import { useState } from 'react';
 import type { FunctionComponent } from 'react';
 
-import Form from './FormWrapper';
 import RuleRow from './RuleRow';
 import AddRuleButton from './AddRuleButton';
+import type { FilterRule } from './types';
 
-const defaultRow = { attribute: '', operator: '', input: '' };
+interface FilterRuleRow extends FilterRule {
+    id: string;
+}
+
+function getNewRow() {
+    const { uuid } = Stage.Utils;
+    const defaultRow: FilterRuleRow = { id: '', type: 'label', key: '', operator: '', values: [] };
+
+    return { ...defaultRow, id: uuid() };
```

Comment:
How about combining the lines together :smile: ?

```suggestion
    return { id: uuid(), type: 'label', key: '', operator: '', values: [] };
```

Then we probably need the type annotation on the function's return (on line 12) :slightly_smiling_face:

Reply:
Don't treat it as final, it will change for sure. I'll leave it as it is.

Reply:
Ok :slightly_smiling_face: